lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -5827,24 +5827,42 @@ void PatchInOutImportExport::exportVertexAttribs(Instruction *insertPos) {
   BuilderBase builder(*m_context);
   builder.SetInsertPoint(insertPos);
 
+  auto resUsage = m_pipelineState->getShaderResourceUsage(ShaderStageFragment);
+  auto &interpInfo = resUsage->inOutUsage.fs.interpInfo;
   for (auto &attribExport : m_attribExports) {
     if (m_gfxIp.major <= 10) {
       unsigned channelMask = 0;
+      unsigned zeroMask = {0}; // The mask of channels with zero value.
       for (unsigned i = 0; i < 4; ++i) {
         assert(attribExport.second[i]);
-        if (!isa<UndefValue>(attribExport.second[i]))
-          channelMask |= (1u << i); // Update channel mask if the value is valid (not undef)
+        if (!isa<UndefValue>(attribExport.second[i])) {
+          const unsigned offset = 1u << i;
+          channelMask |= offset; // Update channel mask if the value is valid (not undef)
+
+          // Check if the valid value is zero and set the mask
+          if (auto channelValue = dyn_cast<ConstantFP>(attribExport.second[i])) {
+            if (channelValue->isZero())
+              zeroMask |= offset;
+          } else if (auto extract = dyn_cast<ExtractElementInst>(attribExport.second[i])) {
+            if (isa<ConstantAggregateZero>(extract->getVectorOperand()))
+              zeroMask |= offset;
+          }
+        }
```

Comment:
This optimization also applies to gfx11+, doesn't it? So the check should be outside the gfxip version check.

Reply:
The corruption only occurs on pre-gfx11 due to the unexpected exp instruction. Does gfx11+ need to do the workaround and optimizaiton? I have planed to set the `workaroundInitializeToZero` in the driver with gfx version check, does it need?

Reply:
I don't know if it needs the workaround, but using DEFAULT_VAL is a possible optimization, since we then can avoid the store instruction for attributes through memory. That is, it is an optimization because memory bandwidth requirements are reduced. That should apply to gfx11 as well.

Reply:
Got it. Thank you for the explanations. Fixed.